### PR TITLE
Support Namespaced Cloud Profiles

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
Due to the deprecation of `Shoot.spec.CloudProfileName`, we now use `Shoot.spec.CloudProfile`, which can reference either a `CloudProfile` or a `NamespacedCloudProfile`. The `GetCloudProfile` method has been updated to handle both types, with their specs accessible via `GetCloudProfileSpec`.

```go
type CloudProfileUnion struct {
	CloudProfile *gardencorev1beta1.CloudProfile
	NamespacedCloudProfile *gardencorev1beta1.NamespacedCloudProfile
}

// CloudProfileUnion returns the CloudProfileSpec of the CloudProfile or NamespacedCloudProfile.
func (u *CloudProfileUnion) GetCloudProfileSpec() *gardencorev1beta1.CloudProfileSpec {
    if u.NamespacedCloudProfile != nil {
        return &u.NamespacedCloudProfile.Status.CloudProfileSpec
    }

    return &u.CloudProfile.Spec
}
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

File: internal/client/garden/client.go [hunk 1]
```diff
@@ -472,30 +473,73 @@ func (g *clientImpl) GetSeedClientConfig(ctx context.Context, name string) (clie
...
 }
 
 // RuntimeClient returns the underlying Kubernetes runtime client.
 func (g *clientImpl) RuntimeClient() client.Client {
 	return g.c
 }
 
-type CloudProfile gardencorev1beta1.CloudProfile
+// AbstractCloudProfile encapsulates a CloudProfile or NamespacedCloudProfile.
+type AbstractCloudProfile struct {
```
Comment (holgerkoser) [inline internal/client/garden/client.go:513]: As discussed rename to `CloudProfileUnion`

**Review by holgerkoser — APPROVED** [review]
/lgtm
Beside from the renaming to `CloudProfileUnion` it is a straight forward change

File: internal/client/garden/client.go (hunk 1 above)
Comment (petersutter) [reply internal/client/garden/client.go]: ✅

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**